weblate/utils/tasks.py

```diff
@@ -83,25 +83,44 @@ def database_backup():
         return
     ensure_backup_dir()
     database = settings.DATABASES["default"]
-    if database["ENGINE"] != "django.db.backends.postgresql":
-        return
-    cmd = ["pg_dump", "--dbname", database["NAME"]]
-    if database["HOST"]:
-        cmd += ["--host", database["HOST"]]
-    if database["PORT"]:
-        cmd += ["--port", database["PORT"]]
-    if database["USER"]:
-        cmd += ["--username", database["USER"]]
-    if settings.DATABASE_BACKUP == "compressed":
-        cmd += ["--file", data_dir("backups", "database.sql.gz")]
-        cmd += ["--compress", "6"]
+    env = get_clean_env()
+
+    if database["ENGINE"] == "django.db.backends.postgresql":
+        cmd = ["pg_dump", "--dbname", database["NAME"]]
+
+        if database["HOST"]:
+            cmd += ["--host", database["HOST"]]
+        if database["PORT"]:
+            cmd += ["--port", database["PORT"]]
+        if database["USER"]:
+            cmd += ["--username", database["USER"]]
+        if settings.DATABASE_BACKUP == "compressed":
+            cmd += ["--file", data_dir("backups", "database.sql.gz")]
+            cmd += ["--compress", "6"]
+        else:
+            cmd += ["--file", data_dir("backups", "database.sql")]
+
+        env["PGPASSWORD"] = database["PASSWORD"]
+    elif database["ENGINE"] == "django.db.backends.mysql":
+        cmd = ["mysqldump", "--result-file", data_dir("backups", "database.sql")]
```

Comment:
I have probably finally found out why the tests do hang. It's not the single backup test, but rather the UI test which triggers it as well (that's why the tests did pass before, I was looking at wrong ones).

The problem is that that test is in the middle of the transaction and mysqldump waits for the transaction to complete and the tests waits for mysqldump, causing nice deadlock.

Disabling the locking on mysqldump seems to address this:

```suggestion
        cmd = [
            "mysqldump",
            "--result-file",
            data_dir("backups", "database.sql"),
            "--single-transaction",
            "--skip-lock-tables",
        ]
```